src/frontends/lean/parser.cpp

```diff
@@ -533,6 +533,10 @@ void parser::add_local_level(name const & n, level const & l, bool is_variable)
 }
 
 void parser::add_local_expr(name const & n, expr const & p, bool is_variable) {
+    if (!m_in_quote) {
+        // certainly not in a pattern
+        m_in_pattern = false;
```

Comment:
@leodemoura I'm not proud of this hack. The issue is that in code like https://github.com/leanprover/lean/blob/d9da6f05b599fb4e1af1bba8b118993eec02f2e5/library/tools/super/prover.lean#L57, it's too late to elaborate the quotation in `patexpr_to_expr`. We need to know whether we're in a pattern before we leave the binder.